src/js/angular/clustermanagement/directives/cluster-list.directive.js

```diff
@@ -196,6 +193,91 @@ function ClusterListComponent($translate, $timeout, $repositories, productInfo,
                 toastr.error(failMessage, $translate.instant('cluster_management.cluster_page.notifications.create_failed'));
             };
 
+            const handlePendingReplacementStrategy = (knownNodesList, endpoint) => {
+                const oldNode = ClusterContextService.getPendingReplace();
+                if (oldNode) {
+                    const node = $scope.cluster.findByEndpoint(knownNodesList, endpoint);
+                    if (node) {
+                        ClusterContextService.replace(oldNode, node);
+                        $scope.editedNodeIndex = undefined;
+                    } else {
+                        const newLocation = RemoteLocationsService.createNewLocation(endpoint);
+                        $scope.setLoader(true, $translate.instant('cluster_management.update_cluster_group_dialog.messages.connecting_node'));
+                        addNewLocation(newLocation)
+                            .then((location) => {
+                                ClusterContextService.replace(oldNode, location);
+                            })
+                            .catch((error) => {
+                                handleErrors(error.data, error.status);
+                                ClusterContextService.emitUpdateClusterView();
+                            })
+                            .finally(() => {
+                                $scope.editedNodeIndex = undefined;
+                                $scope.setLoader(false);
+                            });
+                    }
+                    return true;
```

Comment:
I guess these methods are implemented this way by design, but still I'd like you to confirm if it's OK that they return true/false synchronously, but are executing their job asynchronously

Reply:
Also how about these loaders which these methods are showing. Given that (at least in theory) it's possible to have more than one of them to run at the same time, what happens with the loaders?

Reply:
Also what about the `editedNodeIndex` which can be set by any of them?

Reply:
removed as redundant